packages/foundry/test/GovernedLotteryHook.t.sol

```diff
@@ -0,0 +1,115 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+pragma solidity ^0.8.24;
+
+import "forge-std/Test.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { IVault } from "@balancer-labs/v3-interfaces/contracts/vault/IVault.sol";
+import { GovernedLotteryHook } from "../contracts/hooks/GovernedLotteryHook.sol";
+import { AfterSwapParams, SwapKind } from "@balancer-labs/v3-interfaces/contracts/vault/VaultTypes.sol";
+
+contract GovernedLotteryHookTest is Test {
```

Comment:
This is only testing the proposal and voting stuff; it doesn't actually use the hook. (And couldn't, because there's no `onRegister`, and also `getHookFlags` returns nothing, so the Vault wouldn't call into the hook anyway even if it was connected and registered.

Also, the script is setting the Vault to v2.